src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -130,74 +143,93 @@ public async Task<IActionResult> OnGetCallbackAsync(string returnUrl = null, str
                     i.Email.ToLower() == email.ToLower()
                     && i.Accepted == false);
 
-            // If the user has an invite register them automatically
+            // If the newly logging in User has an unaccepted invite with a matching email address to the User logging
+            // in, register them with the Identity Framework and also create any "internal" User records too if they 
+            // don't already exist.  If they *do* exist already, link the new Identity Framework user with the existing 
+            // "internal" User via the same id (more below).
             if (inviteToSystem != null)
             {
-                var user = new ApplicationUser
+                // Mark the invite as accepted
+                inviteToSystem.Accepted = true;
+                _usersAndRolesDbContext.UserInvites.Update(inviteToSystem);
+
+                // See if we have an "internal" User record in existence yet that has a matching email address to the
+                // new user logging in.  If we do, create this new AspNetUser record with a matching one-to-one id.
+                // Otherwise, later on we'll create a new "internal" Users record with an id matching this AspNetUser's 
+                // id, continuing to establish the one-to-one relationship.
+                var existingInternalUser = await _contentDbContext
+                    .Users
+                    .SingleOrDefaultAsync(u => u.Email.ToLower() == email.ToLower());
```

Comment:
This string comparison if it's executed on the database, which it should be, won't be case sensitive, but probably a good idea to leave the `ToLower` here anyway.